mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -144,4 +144,22 @@ def GENX_BarrierOp : GENX_Op<"barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
+//===----------------------------------------------------------------------===//
+// Subgroup shuffle xor
+
+def GENX_SubGroupShuffleXorOp : GENX_Op<"sub_group_shuffle_xor"> {
```

Comment:
Can you design a generic sub-group shuffle operation in the GENX dialect (similar to NVVM approach) and handle the Xor, up, down operations via an enum. Also make the GENX operation accept different data types for the input args  and the return value. That way the GENX dialect will have the ability to model all the shuffle operations supported by the HW (or at least mirror the API exposed by the SPIRV translator) and provide functionality that is close to the functionality exposed by the NVVM dialect.